projects/saturn/src/main/java/io/fairspace/saturn/webdav/resources/BaseResource.java

```diff
@@ -94,8 +94,19 @@ public Date getCreateDate() {
     }
 
     @Override
-    public final void delete() throws NotAuthorizedException, ConflictException, BadRequestException {
-        var purge = subject.hasProperty(FS.dateDeleted);
+    public void delete() throws NotAuthorizedException, ConflictException, BadRequestException {
+        boolean purge;
+        if (factory.isExtraStoreResource()) {
+            if (subject.hasProperty(RDF.type, FS.ExtraStorageDirectory)) {
+                throw new NotAuthorizedException("Not authorized to purge the extra store root directory.", this, SC_FORBIDDEN);
+            }
+            purge = true;
+        } else {
+            purge = subject.hasProperty(FS.dateDeleted);
+            if (purge && !factory.userService.currentUser().isAdmin()) {
+                throw new NotAuthorizedException("Not authorized to purge the resource.", this, SC_FORBIDDEN);
```

Comment:
why does the non-extraStorageResource path an extra validation + exception, was the old good not suffictient?

Reply:
the validation was there, but I just moved it from `protected void delete(boolean purge)` to `public final void delete()` function, so it is done earlier on and so I can overwrite the protected one in other classes without repeating the validation in all the places.